File: pms-backend/src/main/java/com/pms/service/PrescriptionService.java
package com.pms.service;

import com.pms.dto.PrescriptionDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface PrescriptionService {
    PrescriptionDto createPrescription(PrescriptionDto dto);
    PrescriptionDto updatePrescription(Long id, PrescriptionDto dto);
    PrescriptionDto getPrescriptionById(Long id);
    List<PrescriptionDto> getPrescriptionsByPatientId(Long patientId);
    List<PrescriptionDto> getPrescriptionsByDoctorId(Long doctorId);
    Page<PrescriptionDto> searchPrescriptions(Long patientId, Long doctorId, String query, Pageable pageable);
}

File: pms-frontend/src/services/prescriptionService.js
import api from './api';

export const prescriptionService = {
  searchPrescriptions: async (params = {}) => {
    const res = await api.get('/prescriptions', { params });
    return res.data;
  },

  getPrescriptionById: async (id) => {
    const res = await api.get(`/prescriptions/${id}`);
    return res.data;
  },

  getPrescriptionsByPatientId: async (patientId) => {
    const res = await api.get(`/prescriptions/patient/${patientId}`);
    return res.data;
  },

  createPrescription: async (data) => {
    const res = await api.post('/prescriptions', data);
    return res.data;
  },

  updatePrescription: async (id, data) => {
    const res = await api.put(`/prescriptions/${id}`, data);
    return res.data;
  }
};
